draft-boucadair-add-deployment-considerations.txt

```diff
@@ -181,7 +181,52 @@ Internet-Draft    Discovery of Encrypted DNS Resolvers      October 2022
 
    Unmanaged CPE:  refers to a CPE that is not managed by an ISP.
 
-3.  Sample Target Deployment Scenarios
+3.  DDR/DNR Comparison and Naming Constraints
+
+   DDR requires proving possession of an IP address, as the DDR
+   certificate contains the server's IPv4 and IPv6 addresses and is
+   signed by a WebPKI certificate authority.  DDR is constrained to
+   public IP addresses because WebPKI certificate authorities will not
+   sign special-purpose IP addresses [RFC6890], most notably IPv4
+   private-use [RFC1918], IPv4 shared address [RFC6598], or IPv6
+   Unique-Local [RFC8190] address space.  A tempting solution is to use
+   the CPE's WAN IP address for DDR and prove possession of that IP
+   address.  However, the CPE's WAN IPv4 address will not be a public
+   IPv4 address if the CPE is behind another layer of NAT (either CGN or
+   another on-premise NAT), reducing the success of this mechanism to
+   CPE's WAN IPv6 address.  If the ISP renumbers the subscriber's
+   network suddenly (rather than slow IPv6 renumbering described in
+   [RFC4192]) encrypted DNS service will be delayed until that new
+   certificate is acquired.
+
+   DNR requires proving possession of an FQDN as the encrypted
```

Comment:
The first mechanism is specific to ACME. However, in the second mechanism, the internet-facing server may or may not use ACME (it could use any other certificate management protocol). The bigger challenge with the first mechanism is who will provision the CPE with a unique FQDN. The entity managing the CPE (e.g., ISP) has to provision the CPE with a FQDN; that entity itself can assist the CPE with the certificate. 

In addition, we implemented the second mechanism because of other reasons like (1) No need to host a DNS authoritative server or HTTP server on the CPE (2) Hosting a public-facing HTTP/DNS server is susceptible to outside attacks.  

I don't think the DNR would work on un-managed CPE.

Reply:
> I don't think the DNR would work on un-managed CPE.

DNR does work for that case as well. We do have a specific subsection

Reply:
Yes, but it requires some entity other than the ISP (e..g, security service provider, router vendor) who is managing the encrypted resolver to assign a unique FQDN to the CPE. We can't expect the home administrator or the CPE to get the unique FQDN on its own. Most importantly, it looks infeasible for the CPE to host a DNS authoritative server to complete the ACME challenge.